Detectors/TRD/base/include/TRDBase/LocalGainFactor.h

```diff
@@ -0,0 +1,37 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef O2_TRD_LOCALGAINFACTOR_H
+#define O2_TRD_LOCALGAINFACTOR_H
+
+////////////////////////////////////////////////////////////////////////////////////
+//                                                                                //
+//  TRD calibration class for parameters which are stored pad wise (1.2M entries) //
+//  2019 - Ported from various bits of AliRoot (SHTM)                             //
+//  originally in TRD run2 parlance, this would be a AliTRDCalPad instantiated    //
+//  as a LocalT0                                                                  //
+////////////////////////////////////////////////////////////////////////////////////
+
+#include "TRDBase/PadParameters.h"
+#include "TRDBase/PadCalibrations.h"
+namespace o2
+{
+namespace trd
+{
+
+class LocalGainFactor : public PadCalibrations<unsigned int>
```

Comment:
is the gain integer-valued?

Reply:
@shahor02 oh hell you are right, should be unsigned shorts, will fix. All the templated unsigned int should be unsigned short

Reply:
sorry, still don't understand, why even ``unsigned short``? Aren't these calibrations float-values?

Reply:
The reason why this code is (or soon will be) unsigned short is because the old code was. I honestly never put much thought into it. I just assumed it was values coming from ADCs. Will check and come back with, hopefully, more solid reasoning.

Reply:
Yes
What's stored in the unsigned short is a number < 6.5. So its fixed-point arithmetic [0,65536], [0,6.5].
@tdietel can I remove the fixed-point arithmetic?
I also realised in my fights with the merge this morning the functions with 10k in them have been removed. If I change it to a float it becomes 2.4MB bigger than it currently is as it stores 1.2M values.

Reply:
I thinkg it is ok to use fixed point arithmetics (perhaps it should be wrapped dedicated template which would absorb mapping to float), we can probably even go to ``char`` for corrections lile LocalVDrift.

Reply:
Just chatted to @tdietel the padwise calibrations are run approximately every year, so "wasting" 2.4MB for  each of about 4 or 5 parameters seems not an issue. It was thought when originally doing it, that it would run more frequently and hence the 2.4MB saving each time would be significant.
I will change to float and move on.